controllers/seasons.controllers.ts

```diff
@@ -1,7 +1,116 @@
-// TODO: require the model
-// Season = require("../models/Season")
+import { NextFunction, Response, Request } from "express";
+import Season from "../models/Season";
 
-// TODO: add controller logic 
+function create(req: Request, res: Response, next: NextFunction) {
+  const season = new Season({
+    name: req.body.name,
+    startDate: req.body.startDate,
+    endDate: req.body.endDate, //Think this should be calculated when status is changed to close
+    payroll_timeframe: req.body.payroll_timeframe,
+    price: req.body.price,
+    // TODO: Add product_id, unit_id, currency_id
+  });
 
-// TODO: export module
-// module.exports = { controllerName1, controllerName2, etc};
\ No newline at end of file
+  season
+    .save()
+    .then((result) => {
+      res.status(201).json({
+        message: "Season created",
+        result,
+      });
+    })
+    .catch((error) => {
+      res.status(500).json({
+        message: "Season not created",
+        error,
+      });
+    });
+}
+
+function getAll(req: Request, res: Response, next: NextFunction) {
+  Season.find({ deletedAt: null })
+    .select("name status startDate")
```

Comment:
We should add the endDate as well
<img width="659" alt="Screenshot 2024-02-19 at 9 24 30 AM" src="https://github.com/LangaraVoyagers/project-2-api/assets/57273227/17629a40-0b39-43f1-a4e1-691258b9b490">

Reply:
https://github.com/LangaraVoyagers/project-2-api/pull/8/commits/0a88aeec03f9d2d8bd6d3d0fe65bfda18095ff37